### PR TITLE
[RFT] antfs: new package

### PR DESCRIPTION
**Pull request by dengqf6**
Maintainer: me
Compile tested: mvebu, ramips, ipq40xx
Run tested: ipq40xx

Description:
AVM NTFS is a kernel-space NTFS R/W driver based on libntfs-3g

### CODE REVIEW

File: kernel/antfs/Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://github.com/Sembedded/antfs.git
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_DATE:=2018-08-14
+PKG_SOURCE_VERSION:=002407a1b4f1bc9ec19fb07a4c01fc5348d216a6
+PKG_MIRROR_HASH:=d1c44b2be92531c2caa4fa00b620591f95104436920640e94861f8207d6e9082
+
+PKG_MAINTAINER:=DENG Qingfang <dengqf6@mail2.sysu.edu.cn>
+PKG_LICENSE:=GPL-2.0
```
Comment (neheb) [inline kernel/antfs/Makefile]: GPL-2.0-only I think.

**Comment by neheb** [comment]
@981213 care to comment? I remember you saying you use this.

**Comment by 981213** [comment]
I remembered that the version in the github repository isn't up-to-date and later versions of this driver could be extracted from GPL kernel provided by AVM.

I'm not actually using it. I just built it and tried mounting an NTFS partition. :)

**Comment by klukonin** [comment]
I did a repo with the latest version of code.

https://github.com/klukonin/antfs

According to my limited understanding, patch is not needed for this version =)

Also I made a pull request to the mentioned repo. Lets see what happens.

File: kernel/antfs/Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+PKG_NAME:=antfs
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://github.com/klukonin/antfs.git
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_DATE:=2020-02-10
+PKG_SOURCE_VERSION:=b41ba529f6b64b429527e09a06ce0326d5456c05
+PKG_MIRROR_HASH:=dae039c0fe5bf1a2c8c1cca4211d607a4d6f56fc41b38444e2234b40d710d9db
+
+PKG_LICENSE:=GPL-2.0-only
```
Comment (neheb) [inline kernel/antfs/Makefile]: Based on https://github.com/Sembedded/antfs/blob/master/dir.c , this should be -or-later. No idea if absolutely correct.

Comment (klukonin) [reply kernel/antfs/Makefile]: I think that it's completely correct to use original license.

**Comment by klukonin** [comment]
@dengqf6 

Do you need any help to change the license?

**Comment by klukonin** [comment]
@dengqf6 ping?

**Comment by dengqf6** [comment]
What should I change it to?

**Comment by klukonin** [comment]
@dengqf6 

PKG_LICENSE:=GPL-2.0-or-later

**Comment by klukonin** [comment]
@neheb 
Looks like everything is fine now.

File: utils/antfs-mount/files/mount.ntfs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+mount -t antfs $@
```
Comment (neheb) [inline utils/antfs-mount/files/mount.ntfs]: Use "$@" or "$*"

**Comment by hnyman** [comment]
@dengqf6 @neheb 

antfs-mount fails in some buildbot runs with "dangling symlinks"
(I can't see it actually anywhere as downloadable, so I think that it has never succeeded in buildbot ???)


https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/antfs-mount/compile.txt
```
cp: not writing through dangling symlink '/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/root-x86/./sbin/mount.ntfs'
Makefile:28: recipe for target '/builder/shared-workdir/build/sdk/staging_dir/target-x86_64_musl/root-x86/stamp/.antfs-mount_installed' failed
```

https://downloads.openwrt.org/snapshots/faillogs/arm_cortex-a9_vfpv3-d16/packages/antfs-mount/compile.txt
```
cp: not writing through dangling symlink '/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/root-mvebu/./sbin/mount.ntfs'
Makefile:28: recipe for target '/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/root-mvebu/stamp/.antfs-mount_installed' failed
make[3]: *** [/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/root-mvebu/stamp/.antfs-mount_installed] Error 1
```

**Comment by BKPepe** [comment]
I've just come to this package, and I think that it should be removed as I fail to see why it was added in the first place as the commit description does not say why it was necessary to add it to all targets as now it increased time build, computer resources, etc.

1. It seems that it is NTFS file system by AVM looking at https://github.com/klukonin/antfs/commit/56847418e7e17e3e7f7cfccaf42dad05f9e90105 
   - What it is difference, or why should anyone use it instead of using the two drivers, which are in the Linux kernel? Talking about ``ntfs-3g`` (fuse) and ``ntfs3``
   
2. I could not find this package in any GNU/Linux distributions. 
   - I could not find any webpage or documentation for antfs instead of the source code, which was taken from AVM.
   
All of this points me to the question is antfs maintained and kept up to date? If yes, where?  The first repo is inactive since 2018 and the forked repository was done just to have this merged in OpenWrt. :shrug: 
   - Well, it seems that someone asked the same as I do here https://github.com/openwrt/packages/pull/11154#issuecomment-583828518

And the last question - how many people are using this package at all?

**Comment by BKPepe** [comment]
Also, antfs is not available to Linux kernel 5.15 as it needs to be modified according to https://github.com/openwrt/packages/pull/18398 , so let's drop this package once for all.

**Comment by 981213** [comment]
ntfs3 wasn't a thing when this package was merged. As an in-kernel driver it performs better than ntfs-3g, which is a userspace fuse driver.
Since we have ntfs3 from paragon upstream now, I agree that this package can be retired.

**Pull request merged**